src/nix-eval-jobs.cc

```diff
@@ -294,13 +294,13 @@ int main(int argc, char * * argv)
         auto handler = [&]()
         {
             try {
-                pid_t pid = -1;
                 AutoCloseFD from, to;
 
                 while (true) {
+                    std::optional<Pid> pid;
```

Comment:
Why is this declared inside of the `while` loop?  It seems like this should be declared where the old `pid_t pid` was declared

I'm guessing it's the same behavior even if declared inside the loop, but perhaps a bit confusing because otherwise it look like it is being reinitialized to empty on each iteration of the loop even if that's not actually what's happening

Reply:
Good catch! I think this is actually a bug. 😮